src/ProgressBar.tsx

```diff
@@ -18,18 +20,27 @@ const sizeMap = {
   default: '8px'
 }
 
-const ProgressContainer = styled.span`
+const ProgressContainer = styled.span<
+  {
+    inline?: boolean
+    barSize?: keyof typeof sizeMap
+  } & WidthProps &
+    SystemCommonProps &
+    SxProp
+>`
```

Comment:
What do you think about creating an interface at the top of the file for situations like this where it includes multiple lines? My hunch is that for more complex components like this one it can get a little tricky to visually parse. That seems to be what other applications at GH do, might make it easier to read when jumping between codebases. Curious what @smockle or @T-Hugs thinks

Reply:
Yeah, I agree. I'd love to come up with a naming convention for this kind of props type that's passed to styled-components but not exported and doesn't contain all the props accepted by the returned component. Maybe `Styled[component-name]Props` (e.g. `StyledProgressContainerProps`)?

Reply:
> What do you think about creating an interface at the top of the file for situations like this where it includes multiple lines? My hunch is that for more complex components like this one it can get a little tricky to visually parse.

Great idea! Incidentally, organizing prop types this way helped with the namespace components issue described in https://github.com/primer/components/pull/999#issuecomment-769198110